pkg/scheduler/schedulercache/cache.go

```diff
@@ -112,6 +114,10 @@ func (cache *schedulerCache) UpdateNodeNameToInfoMap(nodeNameToInfo map[string]*
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 	for name, info := range cache.nodes {
+		// Transient scheduler info is added here.
+		if utilfeature.DefaultFeatureGate.Enabled(features.BalanceAttachedNodeVolumes) && info.TransientInfo != nil {
+			info.TransientInfo.TransNodeInfo.reset()
```

Comment:
Shouldn't you allocate `TransientInfo` when it is nil? Also, I think you should be using `resetTransientSchedulerInfo` instead of `reset`.

Reply:
> Shouldn't you allocate TransientInfo when it is nil

I added it initially and then removed it after looking at the code, it seems if we have to reach here, we need should have called SetNode which initializes this struct. 

> I think you should be using resetTransientSchedulerInfo instead of reset.

Yes. Thanks for catching it. Made changes.